### PR TITLE
support new rest api 

### PR DESCRIPTION
**Pull request by gokch**
## What is changed?
support new rest api ( use the same json format as the aergocli )
add getReceipts

## Is there risk of fork?
yes. it need to sync test.

### CODE REVIEW

File: rpc/rpc.go [hunk 1]
```diff
@@ -256,13 +260,21 @@ func (ns *RPC) Receive(context actor.Context) {
...
 			ns.Info().Str("proto", r.Proto).Str("host", r.Host).Stringer("uri", r.URL).Str("content_type", r.Header.Get("content-type")).Str("remote_addr", r.RemoteAddr).Msg("Request handled by other handler. is this correct?")
 			otherHandler.ServeHTTP(w, r)
+			// handler, ok := restAPIHandler(actualServer, r)
+
+			// if(ok) {
+			// 	handler.ServeHTTP(w, r)
+			// }else{
+			// 	ns.Info().Msg("Request handled by other hanlder. is this correct?")
+			// 	otherHandler.ServeHTTP(w, r)
+			// }
```
Comment (kroggen) [inline rpc/rpc.go:277]: are these commented code needed?

File: rpc/web3/web3_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package web3
+
+// TODO : make mock server or remove
```
Comment (kroggen) [inline rpc/web3/web3_test.go:3]: TODO

File: tests/common.sh [hunk 3]
```diff
@@ -38,7 +38,7 @@ wait_version() {
     output=$(../bin/aergocli blockchain 2>/dev/null)
     # check if 'output' is non-empty and starts with '{'
     if [[ -n "$output" ]] && [[ "${output:0:1}" == "{" ]]; then
-      cur_version=$(echo "$output" | jq .ChainInfo.Chainid.Version | sed 's/"//g')
+      cur_version=$(echo "$output" | jq .chainInfo.id.version | sed 's/"//g')
```
Comment (kroggen) [inline tests/common.sh:41]: Why did you change this?
I was expecting backwards compatibility
BTW, it is failing on integration tests and this change can be one of the causes

Comment (kroggen) [reply tests/common.sh:41]: The changes MUST be backwards compatible, otherwise the existing systems will stop working!

Comment (gokch) [reply tests/common.sh:41]: it is change about aergocli response type. ( we change all json key format to camelCase )
can this be a compatibility issue? i thought cli is just viewer, like json api.

Comment (kroggen) [reply tests/common.sh:41]: There are many integration tests on other PRs that are waiting to be merged on V4
This change will make all them stop working
Not good

Comment (kroggen) [reply tests/common.sh:41]: Unless you fix them all later

**Comment by kroggen** [comment]
So this PR only adds a new RPC but does not modify the existing gRPC, right?

So existing SDKs (herajs, heraj, herapy, heraphp, libaergo...) will continue working as expected?

**Comment by kroggen** [comment]
Existing scripts that use aergocli may stop working

I don't know if there are companies using it

**Comment by gokch** [comment]
> So this PR only adds a new RPC but does not modify the existing gRPC, right?
> 
> So existing SDKs (herajs, heraj, herapy, heraphp, libaergo...) will continue working as expected?

yes. existing grpc isn`t changed.

File: tests/common.sh (hunk 3 above)
Comment (gokch) [reply tests/common.sh:41]: sorry about that, but this change is necessary as all keys need to follow the same format.
after merge, I will fix all integration tests in v4 hardfork branch.

Comment (kroggen) [reply tests/common.sh:41]: it is not on a single branch, there are many of them
each PR is on a separate branch

**Comment by gokch** [comment]
> Existing scripts that use aergocli may stop working
> 
> I don't know if there are companies using it

it seems no problem currently. ( I confirmed with hochul, and also request update readthedocs. )

**Comment by kroggen** [comment]
Do you have any example of making requests to the REST API using CURL?

**Comment by gokch** [comment]
> Do you have any example of making requests to the REST API using CURL?

rest api url ( alphanet )
- https://webconnect.blocko.io/swagger
- https://webconnect.blocko.io/v1/getChainInfo

open own port for private network
- add to config.toml
[web3]
netserviceport = 7847
maxlimit = 0
swaggerpath = "."
enable = true

- curl https://localhost:7847/v1/getChainInfo

it will add in readthedocs too.

**Comment by gokch** [comment]
@kroggen can you approve this pr?

File: config/types.go [hunk 4]
```diff
@@ -54,6 +55,14 @@ type RPCConfig struct {
 	NSAllowCORS bool   `mapstructure:"nsallowcors" description:"Allow CORS to RPC or REST API"`
 }
 
+// Web3Config defines configurations for web3 service
+type Web3Config struct {
+	NetServicePort int    `mapstructure:"netserviceport" description:"Web3 service port"`
+	MaxLimit       int    `mapstructure:"maxlimit" description:"Web3 connect linit per second"`
```
Comment (kroggen) [inline config/types.go:61]: linit -> limit

**Review by kroggen — APPROVED** [review]

**Pull request merged**